### DISCUSSION TITLE
Support string format keyword from JSON schema when generating examples

### DISCUSSION THREAD

**Discussion by federicobond** [scalar/scalar#1825] [Ideas]
The format keyword can be useful to generate a more appropriate example for a string schema. For example, for the schema:

```json
{
   "type": "string",
   "format": "email"
}
```
The docs could provide a value of `"user@example.com"` instead of the default example value for strings.

This is what Redocly does in the [Redocly/openapi-sampler](https://github.com/Redocly/openapi-sampler) package.

Check out the JSON Schema [reference](https://json-schema.org/understanding-json-schema/reference/string#format) for more details.

**Comment by marclave** [comment]
@federicobond oh neat! I like this idea, would it be a configuration option where we can autogenerate examples based on the format? like an opt-in improvement? :)

**Reply by hanspagel** [reply]
There are some cases where we don’t want values (e.g. to prefill the request body), but in other cases we want example values (e.g. response examples). I think it’s fine to generate example values for the latter.

https://github.com/scalar/scalar/pull/1866 👀
